CI: daemon/cmd test panic'ed in TravisCI

## CI failure

Hit in https://github.com/cilium/cilium/pull/25221: ~~https://app.travis-ci.com/github/cilium/cilium/builds/262730618#L637-L657~~ (apparently the logs have been overwritten after restarting the job).

```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━���━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓                                                                                                  
┃   PANIC  package: github.com/cilium/cilium/daemon/cmd • Test   ┃                                                                                                  
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛                                                                                                  

panic: send on closed channel                                                                                                                                       

goroutine 787 [running]:                                                                                                                                            
github.com/cilium/cilium/pkg/identity/cache.(*localIdentityCache).lookupOrCreate(0x4001e2a3c0, 0x4000928930, 0x0?)                                                  
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/identity/cache/local.go:109 +0x438                                                                             
github.com/cilium/cilium/pkg/identity/cache.(*CachingIdentityAllocator).AllocateIdentity(0x4001a4b2c0, {0x3425ef8, 0x4000252500}, 0x4000928930, 0x80?, 0x40?)       
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/identity/cache/allocator.go:339 +0x55c                                                                         
github.com/cilium/cilium/pkg/ipcache.(*IPCache).resolveIdentity(0x40005bfe00, {0x3425ef8, 0x4000252500}, {{{0x400254efc0?, 0x0?}, 0x4000114030?}, 0x0?}, 0x0?, 0x0?)
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/ipcache/metadata.go:436 +0xb24                                                                                 
github.com/cilium/cilium/pkg/ipcache.(*IPCache).InjectLabels(0x40005bfe00, {0x3425ef8, 0x4000252500}, {0x400092e9c0, 0x3, 0x3})                                     
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/ipcache/metadata.go:197 +0x4a8                                                                                 
github.com/cilium/cilium/pkg/ipcache.(*IPCache).TriggerLabelInjection.func1({0x3425ef8, 0x4000252500})                                                              
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/ipcache/metadata.go:547 +0x50                                                                                  
github.com/cilium/cilium/pkg/controller.(*Controller).runController(0x4000bbe120)                                                                                   
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/controller/controller.go:216 +0x1f0                                                                           
created by github.com/cilium/cilium/pkg/controller.(*Manager).updateController                                                                                      
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/controller/manager.go:111 +0x8e8                                                                               
FAIL	github.com/cilium/cilium/daemon/cmd	12.711s
```

--- a/pkg/identity/cache/allocation_test.go
+++ b/pkg/identity/cache/allocation_test.go
@@ -398,3 +398,26 @@ func (ias *IdentityAllocatorSuite) TestLocalAllocation(c *C) {
 	mgr.IdentityAllocator.DeleteAllKeys()
 	c.Assert(owner.WaitUntilID(id.ID), Not(Equals), 0)
 }
+
+// Test that we can close and reopen the allocator successfully.
+func (s *IdentityCacheTestSuite) TestAllocatorReset(c *C) {
+	labels := labels.NewLabelsFromSortedList("id=bar;user=anna")
+	owner := newDummyOwner()
+	mgr := NewCachingIdentityAllocator(owner)
+	testAlloc := func() {
+		id1a, _, err := mgr.AllocateIdentity(context.Background(), labels, false, identity.InvalidIdentity)
+		c.Assert(id1a, Not(IsNil))
+		c.Assert(err, IsNil)
+
+		queued, ok := <-owner.updated
+		c.Assert(ok, Equals, true)
+		c.Assert(queued, Equals, id1a.ID)
+	}
+
+	<-mgr.InitIdentityAllocator(nil, nil)
+	testAlloc()
+	mgr.Close()
+	<-mgr.InitIdentityAllocator(nil, nil)
+	testAlloc()
+	mgr.Close()
+}

--- a/pkg/identity/cache/allocator.go
+++ b/pkg/identity/cache/allocator.go
@@ -158,6 +158,13 @@ func (m *CachingIdentityAllocator) InitIdentityAllocator(client clientset.Interf
 		"cluster-id": option.Config.ClusterID,
 	}).Info("Allocating identities between range")
 
+	// In the case of the allocator being closed, we need to create a new events channel
+	// and start a new watch.
+	if m.events == nil {
+		m.events = make(allocator.AllocatorEventChan, eventsQueueSize)
+		m.watcher.watch(m.events)
+	}
+
 	// Asynchronously set up the global identity allocator since it connects
 	// to the kvstore.
 	go func(owner IdentityAllocatorOwner, events allocator.AllocatorEventChan, minID, maxID idpool.ID) {
@@ -212,6 +219,8 @@ func (m *CachingIdentityAllocator) InitIdentityAllocator(client clientset.Interf
 	return m.globalIdentityAllocatorInitialized
 }
 
+const eventsQueueSize = 1024
+
 // InitIdentityAllocator creates the the identity allocator. Only the first
 // invocation of this function will have an effect. The Caller must have
 // initialized well known identities before calling this (by calling
@@ -236,7 +245,7 @@ func NewCachingIdentityAllocator(owner IdentityAllocatorOwner) *CachingIdentityA
 		owner:                              owner,
 		identitiesPath:                     IdentitiesPath,
 		watcher:                            watcher,
-		events:                             make(allocator.AllocatorEventChan, 1024),
+		events:                             make(allocator.AllocatorEventChan, eventsQueueSize),
 	}
 	m.watcher.watch(m.events)
 
@@ -265,6 +274,7 @@ func (m *CachingIdentityAllocator) Close() {
 	m.IdentityAllocator.Delete()
 	if m.events != nil {
 		close(m.events)
+		m.events = nil
 	}
 
 	m.IdentityAllocator = nil

--- a/pkg/identity/cache/cache.go
+++ b/pkg/identity/cache/cache.go
@@ -140,7 +140,6 @@ func (w *identityWatcher) watch(events allocator.AllocatorEventChan) {
 		for {
 			added := IdentityCache{}
 			deleted := IdentityCache{}
-
 		First:
 			for {
 				// Wait for one identity add or delete or stop

